source/physics/include/GateGammaSourceModel.hh

```diff
@@ -0,0 +1,129 @@
+/**
+ *  @copyright Copyright 2017 The J-PET Gate Authors. All rights reserved.
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  @file GateGammaSourceModel.hh
+ */
+
+#ifndef GateGammaSourceModel_hh
+#define GateGammaSourceModel_hh
+
+#include <iostream>
+#include <vector>
+#include "G4PrimaryParticle.hh"
+#include "TLorentzVector.h"
+#include "TVector3.h"
+#include "TRandom3.h"
+#include <GateGammaModelPrimaryParticleInformation.hh>
+
+/** Author: Mateusz Bała
+ *  Email: bala.mateusz@gmail.com
+ *  About class: Clear abstract class which provide source of particles from any process.
+ *  Your task is to write class which inherit this one and write working GetGammaParticles function.
+ *  In addiction this class provide tool to generate linear polarization vectors (or zero polarization vectors).
+ * */
+class GateGammaSourceModel
+{
+ public:
+  GateGammaSourceModel();
+  virtual ~GateGammaSourceModel();
+  /** Function fill each particle with data about momentum.
+   * @param: particles - list with initialized particles - without momentum information
+   * */
+  virtual void GetGammaParticles( std::vector<G4PrimaryParticle*>& particles ) = 0;
+
+  /** Function set positron momentum.
+   * @param: boost - positron boost
+   * */
+  void SetPositronMomentum( const TVector3& boost );
```

Comment:
The dependency to TVector3 (root) instead of G4ThreeVector is a bit annoying because we are currently working to make Gate potentially independent of root. Could you replace TVector3 (and other root dep) by other structures ?